st2actions/st2actions/runners/ssh/paramiko_ssh.py

```diff
@@ -349,6 +349,7 @@ def run(self, cmd, timeout=None, quote=False):
         chan = transport.open_session()
 
         start_time = time.time()
+        chan.get_pty()
```

Comment:
So finally, I got to the bottom of this. Fabric always sets pty=True for sudo operations which is what linux_bootstrap uses. Look at ``stdout`` and ``stderr`` fields carefully in the following logs.

Fabric success: https://gist.github.com/lakshmi-kannan/55eebe65e94ee42cda09
Paramiko success: https://gist.github.com/lakshmi-kannan/ac4131f7de0ff9648650
Paramiko failure: https://gist.github.com/lakshmi-kannan/4d744b2f7938b42dc251

Also a reminder that with pseudo tty enabled, paramiko never failed. Fabric sets pseduo tty for sudo actions because sudo actions take in a password. So setting pseudo tty on sudo actions will solve the problem for us now but I cannot understand why the action still fails. The invoke-rc.d errors in failure case seems fishy. I didn't go anywhere with it. Calling @jfryman and @DoriftoShoes to look at the failure logs. More info posted on slack.